services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -1,18 +1,38 @@
 package cromwell.pipeline.service
 
 import cromwell.pipeline.datastorage.dao.repository.DocumentRepository
+import cromwell.pipeline.datastorage.dto.project.configuration.{
+  ProjectConfigurationRequest,
+  ProjectConfigurationResponse
+}
 import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectId }
 
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ProjectConfigurationService(repository: DocumentRepository)(implicit ec: ExecutionContext) {
   import ProjectConfiguration._
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] =
+  def addConfiguration(configuration: ProjectConfigurationRequest): Future[String] =
     repository
-      .updateOne(toDocument(projectConfiguration), "projectId", projectConfiguration.projectId.value)
+      .updateOne(
+        toDocument(ProjectConfiguration(configuration.projectId, configuration.projectFileConfigurations)),
+        "projectId",
+        configuration.projectId.value
+      )
       .map(_.toString)
 
-  def getById(projectId: ProjectId): Future[Option[ProjectConfiguration]] =
-    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument))
+  def getById(projectId: ProjectId): Future[Option[ProjectConfigurationResponse]] =
+    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument)).map {
+      case Some(conf) if conf.isActive =>
+        Some(ProjectConfigurationResponse(conf.projectId, conf.projectFileConfigurations))
+      case _ => None
+    }
+
+  def deactivateConfiguration(projectId: ProjectId): Future[String] = {
+    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument)).flatMap {
+      case Some(conf) if conf.isActive =>
+        repository.updateOneField("projectId", projectId.value, "isActive", false).map(_.toString)
+      case _ => Future.successful("Activation field doesn't exist")
```

Comment:
How is this even possible that the `activation field` would not exist? :) What does it mean?

Reply:
Done

Reply:
I don't see any changes. Same error message without explanation of how this field may not exists. Perhaps you meant something like `There is no project to deactivate`?